ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamOutput.java

```diff
@@ -71,6 +71,14 @@
    */
   CompletableFuture<DataStreamReply> writeAsync(FilePositionCount src, boolean sync);
 
+  /**
+   * Send out the data in the source file and close stream asynchronously.
+   *
+   * @param src the source file with the starting position and the number of bytes.
+   * @return a future of the reply.
+   */
+  CompletableFuture<DataStreamReply> writeAndCloseAsync(ByteBuffer src);
```

Comment:
Similar to https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#newInputStream-java.nio.file.Path-java.nio.file.OpenOption...- , we may define a WriteOption to include both SYNC and CLOSE
```
diff --git a/ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamOutput.java b/ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamOutput.java
index c054d356..7bd0db98 100644
--- a/ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamOutput.java
+++ b/ratis-client/src/main/java/org/apache/ratis/client/api/DataStreamOutput.java
@@ -41,10 +42,10 @@ public interface DataStreamOutput extends CloseAsync<DataStreamReply> {
    * Send out the data in the source buffer asynchronously.
    *
    * @param src the source buffer to be sent.
-   * @param sync Should the data be sync'ed to the underlying storage?
+   * @param options
    * @return a future of the reply.
    */
-  CompletableFuture<DataStreamReply> writeAsync(ByteBuffer src, boolean sync);
+  CompletableFuture<DataStreamReply> writeAsync(ByteBuffer src, WriteOption... options);
 
 
   /**
diff --git a/ratis-common/src/main/java/org/apache/ratis/io/StandardWriteOption.java b/ratis-common/src/main/java/org/apache/ratis/io/StandardWriteOption.java
new file mode 100644
index 00000000..5e5bc08b
--- /dev/null
+++ b/ratis-common/src/main/java/org/apache/ratis/io/StandardWriteOption.java
@@ -0,0 +1,8 @@
+package org.apache.ratis.io;
+
+public enum StandardWriteOption implements WriteOption {
+  /** Sync the data to the underlying storage. */
+  SYNC,
+  /** Close the data to the underlying storage? */
+  CLOSE
+}
diff --git a/ratis-common/src/main/java/org/apache/ratis/io/WriteOption.java b/ratis-common/src/main/java/org/apache/ratis/io/WriteOption.java
new file mode 100644
index 00000000..9a28d709
--- /dev/null
+++ b/ratis-common/src/main/java/org/apache/ratis/io/WriteOption.java
@@ -0,0 +1,4 @@
+package org.apache.ratis.io;
+
+public interface WriteOption {
+}
```